backend/crates/eiffelvis_core/src/domain/app.rs

```diff
@@ -0,0 +1,51 @@
+use crate::{domain::types::BaseEvent, graph::*};
+use crate::{graph_storage::ChunkedGraph, query::GraphQuery};
+
+use uuid::Uuid;
+
+pub type EiffelGraph = ChunkedGraph<Uuid, BaseEvent, String>;
+
+impl Key for Uuid {}
+
+pub trait EiffelVisApp {
+    fn push(&mut self, event: BaseEvent);
+    fn graph(&self) -> &Self {
+        self
+    }
+    fn get_event(&self, id: Uuid) -> Option<&BaseEvent>;
+    fn dump<'a, T: From<&'a BaseEvent>>(&'a self) -> Vec<T>;
+    fn get_subgraph_with_roots<'a, T: From<&'a BaseEvent>>(&'a self, roots: &[Uuid]) -> Vec<T>;
+}
```

Comment:
This is a bit of an adhoc trait, I believe we can clean this up so that it's not needed anymore but I'm a bit out of ideas right now.